selectrum.el

```diff
@@ -1684,19 +1603,33 @@ suffix."
                                    suffix))))))
             res))))
 
-(defun selectrum--candidates-display-strings (candidates
-                                              highlighted-index
-                                              annot-fun
```

Comment:
The annot-fun is currently not in use but the plan was to have a way to let the insertion style function indicate the current annotation in a different way. For example I imagined I would find it useful for horizontal style showing the annotation of the selected candidate in the modeline.

Reply:
Yes, I understand that you had plans for this and I suspected this. However since it is not currently in use, it is better to remove it and reintroduce it later on. If we do not unnecessarily perform ad-hoc generalizations, we avoid adding cruft. I understand the sentiment to design a flexible API, which is then kept as is. I am convinced that the design with the `annot-fun` and even the slightly refactored design is not good enough to be considered a stable API. It is simply too complicated and entangled. 

My proposal for the mode-line/header-line annotation would be to actually introduce a separate API for this. We could create a function which specifically lets you retrieve the annotation of the current candidate. Then this could be accessed from the display style. I am sure could reuse code with `selectrum--candidates-display-strings`. But let's not do this now, let's do it as soon as we write such a display style. Then one can make a much more informed decision.

Reply:
:+1: